### PR TITLE
Comment delete route

### PR DESCRIPTION
**Pull request by will-j-c**

### CODE REVIEW

File: routes/commentRoutes.js [hunk 1]
```diff
@@ -1,5 +1,26 @@
 const express = require('express');
+const commentController = require('../controllers/commentController');
+const authMiddleware = require('../middlewares/userAuth');
 
 const router = express.Router();
 
+router.post(
+  '/:slug',
+  authMiddleware.isAuthenticated,
+  authMiddleware.isAuthorized,
```
Comment (ZoeNguyen73) [inline routes/commentRoutes.js]: probably won't need authorized? just authenticated should be sufficient for comment posting

File: routes/commentRoutes.js [hunk 2]
```diff
@@ -1,5 +1,26 @@
...
+router.put(
+  '/:id',
+  authMiddleware.isAuthenticated,
+  authMiddleware.isAuthorized,
+  commentController.editComment
+);
+router.delete(
+  '/:id',
+  authMiddleware.isAuthenticated,
+  authMiddleware.isAuthorized,
```
Comment (ZoeNguyen73) [inline routes/commentRoutes.js]: do you want the middleware to handle authorization?
I am still thinking what's the best way to do it. should it be in middleware or should it be in the controller itself...

if we have it in the middleware, then we will need different authorization logic in the file. eg
- for Contributor: authorized user = project owner
- for Comment: authorized user = comment poster
- for User: authorized user = the owner of that profile

File: routes/contributorRoutes.js [hunk 3]
```diff
@@ -4,6 +4,7 @@ const authMiddleware = require('../middlewares/userAuth');
 
 const router = express.Router();
 
+
```
Comment (ZoeNguyen73) [inline routes/contributorRoutes.js]: additional empty space..?

File: routes/commentRoutes.js (hunk 1 above)
Comment (will-j-c) [reply routes/commentRoutes.js]: OK, I agree, have changed.

File: routes/contributorRoutes.js (hunk 3 above)
Comment (will-j-c) [reply routes/contributorRoutes.js]: Removed now

**Review by ZoeNguyen73 — APPROVED** [review]
LGTM!

**Pull request merged**